lib/admin/WP_Auth0_Admin_Appearance.php

```diff
@@ -53,117 +53,91 @@ public function init() {
-      <label for="wpa0_username_style_auto"><?php echo __( 'Auto', 'wp-auth0' ); ?></label>
-
-      <input type="radio" name="<?php echo $this->options->get_options_name(); ?>[username_style]" id="wpa0_username_style_email" value="email" <?php echo esc_attr( $v ) == 'email' ? 'checked="true"' : ''; ?> />
-      <label for="wpa0_username_style_email"><?php echo __( 'Email', 'wp-auth0' ); ?></label>
-
-      <input type="radio" name="<?php echo $this->options->get_options_name(); ?>[username_style]" id="wpa0_username_style_username" value="username" <?php echo esc_attr( $v ) == 'username' ? 'checked="true"' : ''; ?> />
-      <label for="wpa0_username_style_username"><?php echo __( 'Username', 'wp-auth0' ); ?></label>
-
-      <div class="subelement">
-        <span class="description">
-          <?php echo __( 'If you want to allow the user to use either email or password, set it to Auto.', 'wp-auth0' ); ?>
-          <a target="_blank" href="https://auth0.com/docs/libraries/lock/customization#usernamestyle-string-"><?php echo __( 'More info', 'wp-auth0' ); ?></a>
-        </span>
-      </div>
-    <?php
+	public function render_language_dictionary( $args = array() ) {
+		$this->render_textarea_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+		$this->render_field_description(
+			__( 'The languageDictionary parameter for the Auth0 login form. ', 'wp-auth0' ) .
+			sprintf(
+				'<a href="https://github.com/auth0/lock/blob/master/src/i18n/en.js" target="_blank">%s</a>',
+				__( 'List of all modifiable text', 'wp-auth0' )
+			)
+		);
 	}
 
-	public function render_social_big_buttons() {
-		$v = absint( $this->options->get( 'social_big_buttons' ) );
+	public function render_custom_css( $args = array() ) {
+		$this->render_textarea_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+		$this->render_field_description( __( 'Valid CSS to customize the Auth0 login form', 'wp-auth0' ) );
+	}
 
-		echo $this->render_a0_switch( "wpa0_social_big_buttons", "social_big_buttons", 1, 1 == $v );
+	public function render_custom_js( $args = array() ) {
+		$this->render_textarea_field( $args[ 'label_for' ], $args[ 'opt_name' ] );
+		$this->render_field_description( __( 'Valid JS to customize the Auth0 login form', 'wp-auth0' ) );
 	}
 
-	public function render_gravatar() {
-		$v = absint( $this->options->get( 'gravatar' ) );
+	public function render_username_style( $args = array() ) {
+	    $opt_name = $args[ 'opt_name' ];
+	    $id_attr = $args[ 'label_for' ];
+		$value = $this->options->get( $opt_name );
+		$this->render_radio_button( $id_attr . '_au', $opt_name, '', 'Auto', empty( $value ) );
+		$this->render_radio_button( $id_attr . '_em', $opt_name, 'email', '', 'email' === $value );
+		$this->render_radio_button( $id_attr . '_un', $opt_name, 'username', '', 'username' === $value );
+		$this->render_field_description(
+			__( 'To allow the user to use either email or username to login, leave this as "Auto." ', 'wp-auth0' ) .
```

Comment:
Because by reading this I get that "auto" would mean either email or username, but if the db connection has not "requires username = true" then this would only accept email.

Reply:
@lbalmaceda - I was a little confused because the "requires username" setting on the DB connection was taking a while to populate on the Lock side but I see now that Lock does do the right thing even when this is wrong. I'll re-word.